plugins/pkg/file/fs_test.go

```diff
@@ -16,28 +16,44 @@ package file
 
 import (
 	"os"
+	"sync"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestFileMtimeDetection(t *testing.T) {
-	defaultFs = newFS(2000 * time.Millisecond)
+func TestFileIsChanged(t *testing.T) {
+	var (
+		mu      sync.Mutex
+		changed bool
+	)
+	watcher, err := NewWatcher()
+	defer watcher.Stop()
 
-	tmpfile, _ := os.CreateTemp("", "example")
-	defer os.Remove(tmpfile.Name()) // clean up
+	assert.Nil(t, err)
+
+	tmpfile, _ := os.CreateTemp("./", "example")
+
+	file := Stat(tmpfile.Name())
+
+	assert.Equal(t, tmpfile.Name(), file.Name)
 
-	f, err := Stat(tmpfile.Name())
+	err = watcher.AddFile(file)
+	assert.Nil(t, err)
+
+	watcher.Start(func() {
+		mu.Lock()
+		changed = true
+		mu.Unlock()
+	})
 	assert.Nil(t, err)
-	assert.False(t, IsChanged(f))
-	time.Sleep(1000 * time.Millisecond)
 	tmpfile.Write([]byte("bls"))
-	tmpfile.Close()
-	assert.False(t, IsChanged(f))
+	tmpfile.Sync()
 
-	time.Sleep(2500 * time.Millisecond)
-	assert.True(t, IsChanged(f))
-	assert.True(t, Update(f))
-	assert.False(t, IsChanged(f))
+	mu.Lock()
+	assert.True(t, changed)
```

Comment:
We should use waitgroup instead of lock? The test might fail randomly as there is not guaranteed that assert runs after the callback.